apps/src/music/blockly/toolbox/index.ts

```diff
@@ -115,3 +131,165 @@ export function getToolbox(
+  kind: 'category',
+  name: 'Categories',
+  cssconfig: baseCategoryCssConfig,
+  contents: [
+    {kind: 'block', type: BlockTypes.CATEGORY},
+    {kind: 'block', type: BlockTypes.CUSTOM_CATEGORY},
+  ],
+};
+
+/**
+ * Given a toolbox definition and a workspace, iterate through any
+ * categories, add all blocks to the workspace. Used for levelbuilder's
+ * toolbox mode.
+ */
+export function addToolboxBlocksToWorkspace(
+  contents: GoogleBlockly.utils.toolbox.ToolboxItemInfo[],
+  workspace: GoogleBlockly.WorkspaceSvg,
+  clearWorkspace: boolean = true
+) {
+  if (clearWorkspace) {
+    workspace.clear();
+  }
+  contents.forEach(toolboxItem => {
+    if (toolboxItem.kind === 'block') {
+      // Add toolbox blocks directly to the workspace.
+      Blockly.serialization.blocks.append(
+        toolboxItem as GoogleBlockly.serialization.blocks.State,
+        workspace
+      );
+    } else if (toolboxItem.kind === 'category' && 'custom' in toolboxItem) {
+      // For dynamic categories, create a custom category block..
+      const dynamicCategoryName = (
+        toolboxItem as GoogleBlockly.utils.toolbox.DynamicCategoryInfo
+      ).custom;
+      Blockly.serialization.blocks.append(
+        {
+          type: BlockTypes.CUSTOM_CATEGORY,
+          fields: {
+            CUSTOM: dynamicCategoryName,
+          },
+        },
+        workspace
+      );
+    } else if (toolboxItem.kind === 'category') {
+      const categoryInfo =
+        toolboxItem as GoogleBlockly.utils.toolbox.StaticCategoryInfo;
+      // For a localized category, like "Sounds", get the category type, like "Play".
+      const categoryName =
+        localizedNameToCategoryType[categoryInfo.name] ||
+        (categoryInfo.name as Category);
+      if (Category[categoryName]) {
+        // Create a category block
+        Blockly.serialization.blocks.append(
+          {
+            type: BlockTypes.CATEGORY,
+            fields: {
+              CATEGORY: categoryName,
+            },
+          },
+          workspace
+        );
+      } else {
+        // We shouldn't hit this point, but theoretically could if a toolbox
+        // contained unsupported items like buttons or labels.
+        console.warn('Unsupported category found:', toolboxItem);
+      }
+      // Recursively process the contents of the static category
+      addToolboxBlocksToWorkspace(
+        (toolboxItem as GoogleBlockly.utils.toolbox.StaticCategoryInfo)
+          .contents,
+        workspace,
+        false
+      );
+    } else {
+      console.warn('Unsupported toolbox item found:', toolboxItem);
+    }
+  });
+}
+
+/**
+ * Creates a new static category. Used to convert workspace
+ * blocks into a toolbox defintion in levelbuilder's toolbox mode.
+ * @returns JSON representation of a new static category.
+ */
+export function getNewStaticCategory(
+  name: string = 'DEFAULT'
+): GoogleBlockly.utils.toolbox.StaticCategoryInfo {
+  return {
+    kind: 'category',
+    name,
```

Comment:
Does this category name need to be localized?

Reply:
No. This is only for levelbuilder which is English only. In truth, once we move to saving the category name to the id (per your other suggestion), we don't technically need to save it to the name field - except that TypeScript is expecting a string value.